### PR TITLE
[13.0][FIX] stock_picking_import_serial_number: Lot overwriting

### PR DESCRIPTION
**Pull request by ryanc-me**
The import wizard presents an option to "Overwrite Serial", which should
take all serials from the XLSX, and unconditionally write them into existing
SMLs.

However, the condition was not well formed, and so each XLSX line would
write to the same SML - effectively only importing the *last* XLSX line
into only the *first* SML.

### CODE REVIEW

File: stock_picking_import_serial_number/wizard/import_serial_number_wizard.py [hunk 1]
```diff
@@ -78,16 +78,20 @@ def _import_serial_number(self, xl_sheet, stock_move_lines, picking):
         products = self.env["product.product"].search(
             [(self.sn_search_product_by_field, "in", list(product_file_set))]
         )
+        sml_done = self.env["stock.move.line"]
```
Comment (pedrobaeza) [inline stock_picking_import_serial_number/wizard/import_serial_number_wizard.py:81]:
```suggestion
        sml_done = set()
```

File: stock_picking_import_serial_number/wizard/import_serial_number_wizard.py [hunk 2]
```diff
@@ -78,16 +78,20 @@ def _import_serial_number(self, xl_sheet, stock_move_lines, picking):
...
         for item in serial_list:
             product = products.filtered(
                 lambda p: p[self.sn_search_product_by_field] == item[0]
             )
             if picking.picking_type_id.show_reserved:
                 smls = stock_move_lines.filtered(lambda ln: ln.product_id == product)
                 for sml in smls:
-                    if not sml.lot_name or self.overwrite_serial:
+                    lot_ok = not sml.lot_name
+                    overwrite_ok = self.overwrite_serial and sml not in sml_done
```
Comment (pedrobaeza) [inline stock_picking_import_serial_number/wizard/import_serial_number_wizard.py:90]:
```suggestion
                    overwrite_ok = self.overwrite_serial and sml.id not in sml_done
```

File: stock_picking_import_serial_number/wizard/import_serial_number_wizard.py [hunk 3]
```diff
@@ -78,16 +78,20 @@ def _import_serial_number(self, xl_sheet, stock_move_lines, picking):
...
+                    if lot_ok or overwrite_ok:
                         sml.lot_name = item[1]
                         sml.qty_done = 1.0
+                        sml_done += sml
```
Comment (pedrobaeza) [inline stock_picking_import_serial_number/wizard/import_serial_number_wizard.py:94]:
```suggestion
                        sml_done.add(sml.id)
```

**Review by pedrobaeza — COMMENTED** [review]
For being more efficient

**Comment by sergio-teruel** [comment]
@ryanc-me Thanks...I will correct it in the next commit that will also allow selecting serial numbers in internal and ougoing operations.

I will ping you to review

**Comment by rousseldenis** [comment]
@ryanc-me @sergio-teruel What's the status of this ?

**Comment by ryanc-me** [comment]
@rousseldenis the author's PR #1065 supersedes this one

**Comment by pedrobaeza** [comment]
Superseded by #1065

**Pull request closed without merging**